arrow-core-data/src/main/kotlin/arrow/core/ListK.kt

```diff
@@ -185,23 +212,110 @@ data class ListK<out A>(private val list: List<A>) : ListKOf<A>, List<A> by list
         { a, b -> a.some() toT b.some() })
     }
 
+  /**
+   * Returns a [ListK<Tuple2<A?, B?>>] containing the zipped values of the two listKs
+   * with null for padding.
+   *
+   * Example:
+   * ```kotlin:ank:playground
+   * import arrow.core.*
+   *
+   * //sampleStart
+   * val padRight = listOf(1, 2).k().padZip(listOf("a").k())        // Result: ListK(Tuple2(1, "a"), Tuple2(2, null))
+   * val padLeft = listOf(1).k().padZip(listOf("a", "b").k())       // Result: ListK(Tuple2(1, "a"), Tuple2(null, "b"))
+   * val noPadding = listOf(1, 2).k().padZip(listOf("a", "b").k())  // Result: ListK(Tuple2(1, "a"), Tuple2(2, "b"))
+   * //sampleEnd
+   *
+   * fun main() {
+   *   println("padRight = $padRight")
+   *   println("padLeft = $padLeft")
+   *   println("noPadding = $noPadding")
+   * }
+   * ```
+   */
+  fun <B> padZipWithNull(
```

Comment:
Can't we have them use the same name and disambiguate for the jvm using jvmName or something?